### DISCUSSION TITLE
Draw diagram programmatically

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2190] [Q&A]
### Introduction

Hi, is there a way to draw a diagram programmatically (include linking) and move to the desired position, for example to the top-left of paper 

Thanks! 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
Could you please elaborate? What are you trying to do?

```js
// Programmatically position an element
// to the top left corner of the paper
const topLeftCorner = paper.getArea().topLeft();
element.position(topLeftCorner.x, topLeftCorner.y);
```

Are you looking for an automatic layout that will position all the elements and links for you?
See our demos that using automatic layouts [here](https://www.jointjs.com/demos?feature=Automatic+layout).

For instance the [directed graph](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#layout.DirectedGraph) has options to define the `x` and `y` of the resulted graph content (`marginX` and `marginY`).
You can also call the layout and then translate the entire graph to `x` and `y` position with [graph.translate()](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Graph.prototype.translate).
You can crop the paper to show only the graph content with [paper.fitToContent()](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Paper.prototype.fitToContent).

**Reply by pandika-csgi** [reply]
I want to make default diagram, a better option and easiest is load from JSON format like kitchensink did in `sample-graph.ts`
